src/test/fuzz/addrman.cpp

```diff
@@ -114,4 +156,36 @@ FUZZ_TARGET_INIT(addrman, initialize_addrman)
     (void)addr_man.size();
     CDataStream data_stream(SER_NETWORK, PROTOCOL_VERSION);
     data_stream << addr_man;
+
+    // Check that serialize followed by unserialize produces the same addrman.
+
+    data_stream.clear();
+
+    const size_t num_sources = fuzzed_data_provider.ConsumeIntegralInRange<size_t>(1, 100);
+    for (size_t i = 0; i < num_sources; ++i) {
+        const size_t num_addresses = fuzzed_data_provider.ConsumeIntegralInRange<size_t>(1, 1000);
+        std::vector<CAddress> addresses(num_addresses);
+        for (CAddress& addr : addresses) {
+            const auto netaddr = ConsumeNetAddr(fuzzed_data_provider);
+            const auto port = fuzzed_data_provider.ConsumeIntegralInRange<uint16_t>(1, 65535);
+            const auto service = CService{netaddr, port};
+            const auto flags = ServiceFlags{fuzzed_data_provider.ConsumeIntegral<uint64_t>()};
```

Comment:
style-nit: (shouldn't matter here)

```suggestion
            const auto flags = ConsumeWeakEnum(fuzzed_data_provider, ALL_SERVICE_FLAGS);
```

Reply:
Done